pkg/logger/gorm.go

```diff
@@ -1,95 +1,77 @@
-
-		logger.Debugf(sql)
-	} else {
-		gl.logger.Tracef("%v", values[2:]...)
-	}
+func (g gormLogger) LogMode(logger.LogLevel) logger.Interface {
+	// we ignore changes to the log level.
+	// making changes to logger level causes a change to the root logger.
+	return g
+}
+
+func (g gormLogger) Info(ctx context.Context, msg string, args ...interface{}) {
+	g.logger.Infof(msg, args...)
+}
+
+func (g gormLogger) Warn(ctx context.Context, msg string, args ...interface{}) {
+	g.logger.Warnf(msg, args...)
 }
 
-// Code blatantly stolen and modified for our needs from Gorm:
-// https://github.com/jinzhu/gorm/blob/2a3ab99/logger.go#L30-L109
-func formatValues(values []interface{}) string {
-	var formattedValues []string
-	for _, value := range values {
-		indirectValue := reflect.Indirect(reflect.ValueOf(value))
-		if indirectValue.IsValid() {
-			value = indirectValue.Interface()
-			if t, ok := value.(time.Time); ok {
-				if t.IsZero() {
-					formattedValues = append(formattedValues, fmt.Sprintf("'%v'", "0000-00-00 00:00:00"))
-				} else {
-					formattedValues = append(formattedValues, fmt.Sprintf("'%v'", t.Format("2006-01-02 15:04:05")))
-				}
-			} else if b, ok := value.([]byte); ok {
-				if str := string(b); isPrintable(str) {
-					formattedValues = append(formattedValues, fmt.Sprintf("'%v'", str))
-				} else {
-					formattedValues = append(formattedValues, "'<binary>'")
-				}
-			} else if r, ok := value.(driver.Valuer); ok {
-				if value, err := r.Value(); err == nil && value != nil {
-					formattedValues = append(formattedValues, fmt.Sprintf("'%v'", value))
-				} else {
-					formattedValues = append(formattedValues, "NULL")
-				}
-			} else {
-				switch value.(type) {
-				case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64, bool:
-					formattedValues = append(formattedValues, fmt.Sprintf("%v", value))
-				default:
-					formattedValues = append(formattedValues, fmt.Sprintf("'%v'", value))
-				}
-			}
-		} else {
-			formattedValues = append(formattedValues, "NULL")
-		}
+func (g gormLogger) Error(ctx context.Context, msg string, args ...interface{}) {
+	g.logger.WithError(fmt.Errorf(msg, args...)).Errorf(gormLoggerMsg)
+}
+
+func (g gormLogger) Trace(ctx context.Context, begin time.Time, fc func() (string, int64), err error) {
+	sql, rows := fc()
+
+	l := g.logger.WithFields(Fields{
+		gormLoggerTraceFieldKey: Fields{
+			"duration":      time.Since(begin),
+			"affected_rows": rows,
+			"sql":           sql,
+		},
+	})
+
+	if err == nil {
+		l.Tracef("%s", gormLoggerMsg)
+
+		return
 	}
 
-	return strings.Join(formattedValues, " ")
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		l.Tracef(err.Error())
+
+		return
+	}
+
+	l.WithError(err).Errorf(gormLoggerMsg)
 }
 
-func isPrintable(s string) bool {
-	for _, r := range s {
-		if !unicode.IsPrint(r) {
-			return false
-		}
+func (g gormLogger) ParamsFilter(ctx context.Context, sql string, params ...interface{}) (string, []interface{}) {
```

Comment:
I can't find the usage of this function here 🤔  Can we remove it along with changes to the `Logger` interface?

Reply:
Unfortunately it's not documented and I had to extract it from [gorm's source code](https://github.com/go-gorm/gorm/blob/e1f46eb802e7a73c9cc04241c3077dbe9021cd51/callbacks.go#L140).

Logs in `error` level returning error from `sql.Exec`:
* The log with `ParamFilter`
`{"error":"test error","level":"error","message":"gorm DB Logger","sql":{"affected_rows":0,"duration":379690,"sql":"SELECT id FROM users WHERE name = ?"},"timestamp":"2023-02-16T11:33:07Z"}`
* The log without `ParamFilter`:
`{"error":"test error","level":"error","message":"gorm DB Logger","sql":{"affected_rows":0,"duration":321887,"sql":"SELECT id FROM users WHERE name = 'asd'"},"timestamp":"2023-02-16T11:34:01Z"}`

Reply:
Ah OK, I see. But then, if I understand this right, we should never log parameters no matter what log level is it. So, we should just always return `return sql, nil`:

```go
func (g gormLogger) ParamsFilter(ctx context.Context, sql string, params ...interface{}) (string, []interface{}) {
  return sql, nil
}
```

Reply:
I don't understand `we should never log parameters no matter what log level it is` part :sweat_smile: a link or a best practice ?!
As a debuger, I need to see exact query I'm running in my local development, and of course it won't be problematic while no environment level is set to `trace`.
I will leave this as is.

Reply:
So there is a case we needed to set an environment log level to `trace` and that could be problematic, thanks to @Neurostep for pointing that out.
I reverted it back, please have another look :pray: